### PR TITLE
[ConstraintSystem] Remove implementation of operator designated types in the solver.

### PR DESCRIPTION
**Pull request by hborla**
Remove the `-solver-enable-operator-designated-types` flag and implementation in the solver. This experimental feature is no longer being explored, and we're implementing other approaches to improving constraint solver performance that break this feature.

There's a little more to remove (the `-enable-operator-designated-types` flag enables designated types in the parser, and designated type syntax is adopted in the standard library) but this lets me make more progress on chipping away at solver performance hacks.

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

File: lib/Sema/CSSolver.cpp [hunk 1]
```diff
@@ -2320,17 +2155,6 @@ Constraint *ConstraintSystem::selectDisjunction() {
...
     return nullptr;
 
-  // Attempt apply disjunctions first. When we have operators with
-  // designated types, this is important, because it allows us to
-  // select all the preferred operator overloads prior to other
-  // disjunctions that we may not be able to short-circuit, allowing
-  // us to eliminate behavior that is exponential in the number of
-  // operators in the expression.
-  if (getASTContext().TypeCheckerOpts.SolverEnableOperatorDesignatedTypes) {
-    if (auto *disjunction = selectApplyDisjunction())
```
Comment (xedin) [inline lib/Sema/CSSolver.cpp:2330]: I think `selectApplyDisjunction` could also be removed now right?

Comment (hborla) [reply lib/Sema/CSSolver.cpp:2330]: Ah you're right, thanks for catching that!

**Review by xedin — APPROVED** [review]
LGTM! I left a comment inline about `selectApplyDisjunction`.

**Comment by hborla** [comment]
@swift-ci please smoke test

**Pull request merged**